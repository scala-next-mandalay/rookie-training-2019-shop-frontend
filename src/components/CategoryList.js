import React from 'react'
import PropTypes from 'prop-types'
import { Box, List, ListItem, ListItemText } from '@material-ui/core'
import './style.css'

const CategoryList = ({ categories, setCategoryId, handleDrawerClose }) => {
  return (
    <List>
        {categories.map((obj) => (
          <ListItem button key={obj.id} onClick={()=>{
           handleDrawerClose()
            setCategoryId(obj.id)
          }}  >
            <ListItemText>
            <Box fontWeight={600} color='#9A7B66' fontStyle='italic' class='ho'>
              {obj.name}
            </Box>
            </ListItemText>
          </ListItem>
        ))}
      </List>
  )
}

CategoryList.propTypes = {
  categories: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired
  }).isRequired).isRequired,
  setCategoryId: PropTypes.func,
  handleDrawerClose: PropTypes.func,
}

export default CategoryList
