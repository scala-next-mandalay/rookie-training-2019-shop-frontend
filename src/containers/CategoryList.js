import { connect } from 'react-redux';
import CategoryList from '../components/CategoryList';
import { setCategoryId } from '../modules/items';

export default connect(
  (state) => ({
    categories: state.categories.rows,
  }),
  (dispatch) => ({
    setCategoryId: (id) => dispatch(setCategoryId(id)),
  })
)(CategoryList);
