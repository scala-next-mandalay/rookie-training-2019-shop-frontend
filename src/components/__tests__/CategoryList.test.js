import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Parent from '../Parent';
import { act } from 'react-dom/test-utils';
import CategoryList from '../CategoryList';
/*global expect*/
/*global jest*/
/*global MouseEvent*/

let container;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.outerHTML = '<body></body>';
  container = null;
});

const handleDrawerClose= jest.fn(() => {

})
let categories = [{ id: 1, name: "category1" },
                  { id: 3, name: "category3" }];
let id=null;
const setCategoryId = jest.fn((value) => {
  id = value;
});

describe("CategoryList component", () => {
  it('matches the snapshot', () => {
    const CategoryListSnapshot = renderer.create(
      <Parent><CategoryList 
      setCategoryId={setCategoryId} 
      categories={categories}
      handleDrawerClose={handleDrawerClose}/></Parent>
    ).toJSON();
    expect(CategoryListSnapshot).toMatchSnapshot();
  });
})

describe("testing category list", () => {
      it('Testing category data show', () => {
        let categories = [ {id:1 , name : "category1"},{id :3, name: "category3"}];
        
        act(() => {
          ReactDOM.render((
            <Parent>
              <CategoryList handleDrawerClose={handleDrawerClose} categories={categories} setCategoryId={setCategoryId}/>
            </Parent>
          ), container);
         
          
       const spanDivArr = document.querySelectorAll('span div');
         act(() =>{
            spanDivArr[0].dispatchEvent(new MouseEvent('click', {bubbles: true}));
          });
          expect(setCategoryId).toHaveBeenCalled();
          expect(id).toBe(1);
          
          act(() =>{
            spanDivArr[1].dispatchEvent(new MouseEvent('click', {bubbles: true}));
          });
          expect(setCategoryId).toHaveBeenCalled();
          expect(id).toBe(3);
          expect(spanDivArr[1].textContent).toBe("category3");
        });
      });
});
      