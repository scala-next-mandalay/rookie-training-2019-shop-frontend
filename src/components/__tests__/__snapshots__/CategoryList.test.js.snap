// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryList component matches the snapshot 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        <div
          className="MuiBox-root MuiBox-root-56 ho"
        >
          category1
        </div>
      </span>
    </div>
  </div>
  <div
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        <div
          className="MuiBox-root MuiBox-root-57 ho"
        >
          category3
        </div>
      </span>
    </div>
  </div>
</ul>
`;
